### ISSUE TITLE
Ability to produce predictions based on given user id

### ISSUE THREAD

**Issue by vincent0408** [fleanend/TorchGlocalK#1]
Hi, I was wondering if the trained model is capable of generating k recommendations by just giving the user id. All the user ids given are already seen in the MovieLens-100k dataset. If this is possible, could you point me a direction on how to generate the results?

**Comment by fleanend** [comment]
Hi!
It's possible, actually it's the only way to get recommendations.
After the fine-tuning stopped you can call:

> pre, _ = complete_model(torch.Tensor(train_r).double().to(device))
> pre = pre.float().cpu().detach().numpy()

*pre* is, thus, the reconstructed rating matrix.

If you have *R* as the original rating matrix, you can filter out the previously seen movies from the reconstructed matrix like so

>mask = np.greater(R, 1e-12).astype('float32')
>recommendations = pre*(1-mask)

Then you can do argsort on recommendations[user_index] to find the best movies for that user!

Hope this answers your question, I will add a working example asap, thanks for the question!

**Comment by vincent0408** [comment]
Wow thanks for the detailed response, I do have a dumb, small question though, how do I get R the original rating matrix? From the code I assume train_r, test_r are the rating matrix for training and testing data, so should the original rating matrix R be computed from the original movielens-100k dataset?

**Comment by fleanend** [comment]
No problem!
The original R should be the complete dataset you get from movielens-100k dataset, which is equal to train_r+test_r

In my snippet in the previous answer I wrote train_r where you should use R for actual inference, as my code is focused only on benchmarking the results.

**Comment by vincent0408** [comment]
Thanks for the reply, all is clear now!
